libs/libc/stdio/lib_stdio_cb.c

```diff
@@ -0,0 +1,76 @@
+/****************************************************************************
+ * libs/libc/stdio/lib_stdio_cb.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "libc.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: lib_fseek_cb
+ ****************************************************************************/
+
+off_t lib_fseek_cb(FAR void *s, off_t offset, int whence)
+{
+  FAR FILE *stream = (FAR FILE *)s;
+  return lseek(stream->fs_fd, offset, whence);
+}
+
+/****************************************************************************
+ * Name: lib_fwrite_cb
+ ****************************************************************************/
+
+ssize_t lib_fwrite_cb(FAR void *s, const char *buf, size_t size)
+{
+  FAR FILE *stream = (FAR FILE *)s;
+  return _NX_WRITE(stream->fs_fd, buf, size);
```

Comment:
change _NX_WRITE to write

Reply:
_NX_WRITE is either write or nx_write based on which mode we are in and whether this is kernel space or user space. STDIO for protected build should be compiled twice, once for kernel and once for user space, therefore macro should decide which function is called.

Reply:
stdio shouldn't be called inside kernel space, so _NX_WRITE isn't needed.

Reply:
Sorry for the delay @xiaoxiang781216, I was on a vacation without notebook.

NuttX documentation states

> Normally the logic in this file builds to a single library (libc.a). However, if NuttX is built as a separately compiled kernel (with CONFIG_BUILD_PROTECTED=y or CONFIG_BUILD_KERNEL=y), then the contents of this directory are built as two libraries: One for use by user programs (libc.a) and one for use only within the <kernel> space (libkc.a).

Therefore stdio can be called inside kernel space and `_NX_WRITE` should be there to decide.

Reply:
Ok, let's keep it.